CONTRIBUTING.md

```diff
@@ -0,0 +1,46 @@
+# Contributing to parity-common
+
+parity-common welcomes contribution from everyone in the form of suggestions, bug
+reports, pull requests, and feedback. This document gives some guidance if you
+are thinking of helping us.
+
+Please reach out here in a GitHub issue if we can do anything to help you contribute.
+
+## Submitting bug reports and feature requests
+
+When reporting a bug or asking for help, please include enough details so that
+the people helping you can reproduce the behavior you are seeing. For some tips
+on how to approach this, read about how to produce a [Minimal, Complete, and
+Verifiable example].
+
+[Minimal, Complete, and Verifiable example]: https://stackoverflow.com/help/mcve
+
+When making a feature request, please make it clear what problem you intend to
+solve with the feature, any ideas for how parity-common could support solving that problem, any possible alternatives, and any disadvantages.
+
+## Versioning
+
+As many crates in the rust ecosystem, all crates in parity-common follow [semantic versioning]. This means bumping PATCH version on bug fixes that don't break backwards compatibility, MINOR version on new features and MAJOR version otherwise (MAJOR.MINOR.PATCH). Versions < 1.0 are considered to have the format 0.MAJOR.MINOR, which means bumping MINOR version for all non-breaking changes.
```

Comment:
Worth mentioning the exception of `serde` that isn't following semver? Caution against `rand` upgrades?

Reply:
Not sure what to add here. Although `serde` doesn't follow semver for `minor` and `patch` bumping, it will bump `major` on a breaking change. And not sure how `rand` is different in terms of breaking changes from other crates. But open to any suggestions.

Reply:
`serde`: so you're saying we don't need to mention it not following semver because `minor/`patch` releases are backwards compatible?
`rand`: my thinking was simply that some crate updates are (much) more risky than others, but it's hard to come out and say "don't send PRs to upgrade `rand` because it's a mess". But I guess we can handle that as/if PRs come in.

Reply:
yep, pretty much